app/components/WalletProvider.tsx

```diff
@@ -11,23 +11,25 @@ import { infuraProvider } from "wagmi/providers/infura";
 export default function WalletProvider({ children, network }: { children: ReactElement; network: string }) {
   let configureChainObj;
 
-  //TODO create config of all chains we support
-
-  // By adding defaultChains to all instances, we avoid the message when on the wrong network of:
-  // "switch to chain {chain.id}" instead of chain name. The ID is used when the currently connected chain
-  // is not in our configured chains. We don't have to stick with this set up but this will allow most chains to
-  // show the proper name.
   const infuraId = "54fcc811bac44f99b84a04a4a3e2f998";
-  if (network === "ropsten") {
-    configureChainObj = configureChains(defaultChains, [infuraProvider({ apiKey: infuraId }), publicProvider()]);
-  } else if (network === "polygon") {
-    configureChainObj = configureChains([...defaultChains, chain.polygon], [infuraProvider({ apiKey: infuraId }), publicProvider()]);
-  } else if (network === "localhost") {
-    configureChainObj = configureChains([...defaultChains, chain.polygon, chain.hardhat, chain.localhost], [publicProvider()]);
-  } else {
-    //nothing provided will lead to front end error
-    configureChainObj = configureChains(defaultChains, [publicProvider()]);
-  }
+  configureChainObj = configureChains(
+    [...defaultChains, chain.polygon, chain.hardhat, chain.localhost], 
```

Comment:
I believe if you do it this way it will try to connect to those default chains - and find contracts for those chains if a user is on that network no?